core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1364,6 +1376,46 @@ private ExprValue exprQuarter(ExprValue date) {
     return new ExprIntegerValue((month / 3) + ((month % 3) == 0 ? 0 : 1));
   }
 
+  /**
+   * Returns TIME value of sec_to_time function for an INTEGER or LONG arguments.
+   * @param totalSeconds The total number of seconds
+   * @return A TIME value
+   */
+  private ExprValue exprSecToTime(ExprValue totalSeconds) {
+    return new ExprTimeValue(LocalTime.MIN.plus(Duration.ofSeconds(totalSeconds.longValue())));
+  }
+
+  /**
+   * Helper function which obtains the decimal portion of the seconds value passed in.
+   * Uses strings to prevent issues with math on floating point numbers.
+   * Return is formatted to be used with Duration.ofSeconds();
+   *
+   * @param seconds and ExprDoubleValue or ExprFloatValue for the seconds
+   * @return A LONG representing the nanoseconds portion
+   */
+  private long formatNanos(ExprValue seconds) {
+    String nanoStringVal = seconds.doubleValue().toString();
+
+    //Obtains only the decimal part of the seconds ExprVal
+    String fraction = nanoStringVal.substring(nanoStringVal.indexOf(".") + 1);
+    fraction = StringUtils.rightPad(fraction, 9, "0");
+    long formattedNanos = Long.valueOf(fraction);
```

Comment:
Is it possible to avoid conversion `double -> string -> long`?
For example:
```java
value = value - Math.floor(value);
value *= 10^9;
value = value - Math.floor(value);
return Math.round(value); // to convert double to long
```

Reply:
I've found it to be very difficult and unreliable no matter what method I use to keep it as a number. I've tried to do this math a few different ways, but there are always issues with imprecision when it comes to the step of multiplying by 10^9. I thought that it might be better to convert it to string even if there is potentially a performance hit because it guarantees a correct answer, but maybe it just requires an entirely different way to do the math than passing it into the `Duration.ofSeconds()` function. Any suggestions?

Reply:
```java
    var d = Double.valueOf(507116.921083338D);
    d = (d - Math.floor(d)) * 1_000_000_000;
    var r = Double.valueOf(Math.floor(d)).longValue();
```

Reply:
Altered implementation to use `BigDecimal` instead. https://github.com/Bit-Quill/opensearch-project-sql/pull/219/commits/3064a16f34af30b0908b10fd74dfc3769e5bc0cb